### ISSUE TITLE
Monte Carlo Commandline: TallyCount in ROfFx detector larger than N

### ISSUE THREAD

**Issue by hayakawa16** [VirtualPhotonics/VTS#7]
A Boston University user reports she has produced results in which the TallyCount of the ROfFx detector is larger than the number of photons launched, N.

**Comment by hayakawa16** [comment]
Found a bug in ROfFxDetector.  The incrementing of TallyCount was occurring within the fx loop so that the resulting output was (number of fx) x (actual TallyCount). 
Moved the incrementing line outside of loop.

**Comment by hayakawa16** [comment]
Since TallyCount was not part of any unit tests, I have now added the output of TallyCount to SimulationOutput for unit testing.  I added unit tests for all detectors and found that T(fx) and Flu(x,y,z,omega) calculated TallyCount in error.  I also found an inconsistency in the detector names in SimulationOutput Flu(x,y,z,omega) used to be called flu_xyzo, however, the rest of the detectors with omega dependence used "w" as the symbol for omega or flu_xyzw.  So this output change was made.
